File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int start = 0;
        int end = arr.length-1;
        
        while(start<end){
            int mid = start + (end - start)/2;
            int mid2 = mid+1;
            if(arr[mid]<arr[mid2]){
                start = mid2;
            }else{
                end = mid;
            }
        }
        return start;
    }
}

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int low = 0;
        int high = arr.size()-1;
        while(low<high){
            int mid = (low+high)/2;
            int mid2 = mid+1;
            if(arr[mid]<arr[mid2]){
                low = mid2;
            }else{
                high = mid;
            }
        }
        
        return low;
    }
};